File: Library/JumpGame.java
import java.util.*;

class JumpGame {
    public boolean canJump(int[] nums) {
        int m=0;
        for(int i=0; i<nums.length; i++) {
            if(i>m) return false;
            m=m<i+nums[i] ?i+nums[i] :m;
            if(m>=nums.length-1) return true;
        }
        return true;
    }
}

File: neetcode/JumpGame.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool canJump(vector<int>& nums) {
        int max_r=nums[0];
        for(int i=1; i<nums.size(); ++i) {
            if(max_r<i) return false;
            if(max_r<i+nums[i]) max_r=i+nums[i];
        }
        return true;
    }
};
